### DISCUSSION TITLE
How to obtain atomic information such as radius, number, and the number of valence electrons and protons?

### DISCUSSION THREAD

**Discussion by sunml99** [deepmodeling/abacus-develop#662] [Q&A] [answered]
## Describe Current Status and Possible Solution

We are trying to implement the implicit solvation model in ABACUS. Now we need the atomic radius, atomic number, and the number of valence electrons and protons of the atom. 
Has the functions of giving the atomic type to get these parameters been implemented in the code of ABACUS? How to call them?

**Comment by dyzheng** [comment, marked as answer]
Thanks for your question!
In ABACUS, we can find atomic information come from the function below:
https://github.com/deepmodeling/abacus-develop/blob/098b6177a430283f3feca2eb20f70b46df4e188e/source/module_cell/read_pp.cpp#L56
In this function, information for element is imported from pseudopotential files.
For example: 
* atomic radius is "GlobalC::ucell.atoms[it].r[GlobalC::ucell.atoms[it].mesh-1]", unit is bohr;
* atomic number as same as protons of the atom is not in ABACUS, you can refer to periodic table with "GlobalC::ucell.atoms[it].psd";
* number of valence electrons is "GlobalC::ucell.atoms[it].zp".
